samples/requesterPays.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This application demonstrates how to perform basic operations on buckets with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+
+'use strict';
+
+function enableRequesterPays(bucketName) {
+  // [START storage_enable_requester_pays]
+  // Imports the Google Cloud client library
+  const Storage = require(`@google-cloud/storage`);
```

Comment:
This might be old-school JS conventions, but I think CaptialCase was meant for constructors, thus you have to use `new` in front of them. The tooling and style linters have come a long way, so ignore me if they ignore it now, too.

Reply:
The way our libraries are implemented, `Storage` _is_ a constructor. `new Storage()` works just fine. If I had documented the client libraries, I would have used `new` when creating `Storage` instances.

Consider this:

```js
// what is "storage"?
const storage = require('@google-cloud/storage');

// oh, I guess it's a factory function, which gives me...something
const client = storage();
```
and compare to:
```js
// Capital "Storage", must be a class
const Storage = require('@google-cloud/storage');

// Okay, so I make a new instance
const storage = new Storage();
storage instanceof Storage; // true

// And if it's a class, I ought to be able to do this
class MyStorage extends Storage {
  getBuckets(options, callback) {
    // do something custom
    return super.getBuckets(options, callback);
  }
}
```

If/when we switch to ES2015 classes, then the class constructor will no longer be able to serve as a factory function:

```js
// TypeError: Class constructor Storage cannot be invoked without 'new'
const storage = Storage();
```

Reply:
I'm with you on all points, but I'm looking at this in the examples I've found:

```js
// Instantiates a client
const storage = Storage({
  projectId: projectId,
});
```

So, there's just no `new`, and I think that's what we want?